### PR TITLE
parser: support check constraints field

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: Arenatlx <314806019@qq.com>

### What problem does this PR solve? 
support check constraint info.

### What is changed and how it works?
1：add constraint info in model.go
2：add a item to store the check constraint info in tableInfo.

### Check List 

Tests 

 - Integration test

Code changes

 - Has changed the model

Related changes

 - Need to update the documentation

### CODE REVIEW

File: ast/ddl.go [hunk 1]
```diff
@@ -477,6 +477,8 @@ type ColumnOption struct {
 	AutoRandomBitLength int
 	// Enforced is only for Check, default is true.
 	Enforced bool
+	// Name is only used for Check Constraint name.
+	Name string
```
Comment (tangenta) [inline ast/ddl.go:481]: How about renaming it to `ConstraintName`?

Comment (AilinKid) [reply ast/ddl.go:481]: good

**Comment by AilinKid** [comment]
this pr is ready to merge PTAL @tangenta

File: ast/ddl.go (hunk 1 above)
Comment (AilinKid) [reply ast/ddl.go]: addressed

**Comment by zz-jason** [comment]
@AilinKid please create an issue for this PR and link the issue to the project kanban.

**Comment by AilinKid** [comment]
> @AilinKid please create an issue for this PR and link the issue to the project kanban.

ok

**Comment by AilinKid** [comment]
Linked Issue: https://github.com/pingcap/tidb/issues/17402

**Pull request merged**